src/message_dao_test.py

```diff
@@ -0,0 +1,115 @@
+import unittest
+import subprocess
+import zlib
+
+from unittest import mock
+from google.cloud.sql.connector import Connector
+
+import message
+import message_dao
+import archive_converter
+from test_helpers import test_data_path
+
+class MessageDaoTest(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(mock.patch.stopall)
+        self.mock_connect = mock.patch.object(Connector, 'connect').start()
+        mock_connection = self.mock_connect.return_value
+        self.mock_commit = mock_connection.commit
+        self.mock_cursor = mock_connection.cursor.return_value.__enter__.return_value
+        self.mock_execute = self.mock_cursor.execute
+        # Check initialization of DAO
+        self.dao = message_dao.MessageDao('FAKE_GIT_PATH')
+        self.mock_connect.assert_called_once()
+        self.mock_commit.assert_called_once()
+        self.assertEqual(3, self.mock_execute.call_count)
+        self.reset_mocks(return_value=True, side_effect=True)
+        self.mock_connect.side_effect = RuntimeError("Shouldn't be called after init")
+
+    def reset_mocks(self, return_value=False, side_effect=False):
+        self.mock_connect.reset_mock(return_value=return_value, side_effect=side_effect)
+        self.mock_execute.reset_mock(return_value=return_value, side_effect=side_effect)
+        self.mock_commit.reset_mock(return_value=return_value, side_effect=side_effect)
+
+    def test_store(self):
+        # Testing storing Message that's not already in DAO 
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        sql_text = "REPLACE INTO Messages VALUES (%s, %s, %s, %s, %s, %s, %s)"
+        self.dao.store(email)
+        self.mock_execute.assert_called_once_with(sql_text, mock.ANY)
+        self.mock_commit.assert_called_once()
+        self.reset_mocks()
+
+        # Testing storing Message that's already in DAO 
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
```

Comment:
this extra test case doesn't do anything, right?

i.e. it doesn't test any new codepaths or different state, since it's all mocked out.
we should drop this block and the reset_mocks() call above it.

Reply:
Removed extra test case since code coverage does remain the same